index.js

```diff
@@ -4,9 +4,57 @@ let form = document.querySelector("form");
 form.addEventListener("submit", function (event) {
   event.preventDefault();
 
-  console.log("Form submitted");
+  const maxHoursPerLevel = {
+    basic: 5,
+    advanced: 10,
+  };
+
+  let userEmail = document.querySelector("#email").value;
+
+  let userLevel = document.querySelector("#level").value;
+
+  let userHours = parseInt(document.querySelector("#hoursPerWeek").value);
+
+  // Validate the user's input
+  let errors = {};
+
+  // Helper function to add error messages
+  function addError(field, message) {
+    if (!errors[field]) {
+      errors[field] = { messages: [] };
+    }
+    errors[field].messages.push(message);
+  }
+
+  // Check if the user has provided an email address
+  if (userEmail === "") {
+    addError("email", "Please enter your email address.");
+  }
+
+  if (userLevel === "") {
+    addError("level", "Please select a level of study.");
+  }
+
+  if (isNaN(userHours) || userHours < 1) {
+    addError("hoursPerWeek", "Please enter at least one hour of tuition.");
+  }
+
+  if (!maxHoursPerLevel.hasOwnProperty(userLevel)) {
+    addError("level", "Invalid level of study selected.");
+  }
+
+  const maxAllowedHours = maxHoursPerLevel[userLevel];
+  if (userHours > maxAllowedHours) {
+    addError(
+      "hoursPerWeek",
+      `You can only study a maximum of ${maxAllowedHours} hours per week.`
+    );
+  }
+
+  console.log({ errors });
```

Comment:
Good start. You are now collected each error message into a single object that we can use to display all error messages at once.

It looks as though you are now ready to [style and display these messages to the user](https://gedsargent.github.io/code-monster/guides/programming/harmony-music/error-handling/#style-input-fields-with-errors).